src/huggingface_hub/repository.py

```diff
@@ -631,6 +631,9 @@ def clone_from(self, repo_url: str, use_auth_token: Union[bool, str, None] = Non
                 repo_url += REPO_TYPES_URL_PREFIXES[self.repo_type]
 
             if token is not None:
+                if token.startswith("api_org"):
```

Comment:
Not that an org token cannot be used here because:

```py
valid_organisations = [org["name"] for org in whoami_info["orgs"]]
```

would fail with a bad error message => let's catch it early with a better error message for the user

Reply:
Small change but I would move those 2 lines at the beginning of the method just after the other token check (`if token is None and self.private:` [line 616](https://github.com/huggingface/huggingface_hub/blob/7a869774b95166629fb44af58afae01b4d5fe5bd/src/huggingface_hub/repository.py#L616)) so that both token validations are handled in the same place.